### ISSUE TITLE
Multiple Responses aren't supported

### ISSUE THREAD

**Issue by rjocoleman** [apiaryio/dredd#25]
It appears that multiple responses aren't supported. This limits the usefulness of `dredd` when using something like `grunt` or `guard`.

My use case is creating a subscriber, when a subscriber is created successfully 200. When it's a duplicate it's a 422. First test will create the subscriber and all subsequent tests will fail

```
$ dredd apiary.apibp http://localhost:3000/
INFO: Runtime compilation warning: Multiple responses, using first.
 on  > Subscribers > Creating a Subscriber
```

**Comment by arvenil** [comment]
Any plans for this?

**Comment by netmilk** [comment]
Please refer to this [comment](https://github.com/apiaryio/dredd/issues/88#issuecomment-70084799). I will create some example on this, definitely soon.

**Comment by GregorioDiStefano** [comment]
Is there anyway to use a before hook multiple times on the same name/api call?

I have an API that has several different responses, depending on the request, and I'm trying to get dredd to test each response possibility.

I can modify my request using a before hook, but multiple before hooks on the same hook are modify the single request.

**Comment by netmilk** [comment]
@GregorioDiStefano If you have given multiple request and responses under one action, It should compile to multiple examples and you should be able to address them in hooks. See output of `--names` 

Would you mind sharing some snippet of your API Blueprint with me?

**Comment by netmilk** [comment]
The discussion about how to address multiple request and responses in the future is being discussed here: https://github.com/apiaryio/dredd/issues/227#issuecomment-174529415

**Comment by GregorioDiStefano** [comment]
This is what I'm working with:

### Devices in deplyment [GET /api/0.0.1/deployments/{deployment_id}/devices]
Device statuses for the deployment.
​
+ Parameters
    + deployment_id: `f81d4fae-7dec-11d0-a765-00a0c91e6bf6` (string,required) - Deployment identifier
​
+ Response 200 (application/json)
    + Attributes (array)
        + attribute (DeploymentStatic)
            + id: `00a0c91e6-7dec-11d0-a765-f81d4faebf6` (string) - Device identifier
            + status: `pending` (enum[string]) - Deployment status
                + Members
                    + `pending` - Pending to start
                    + `inprogress` - In progress
                    + `success` - Finished with success
                    + `failure` - Finished with failure
                    + `noimage` - Do not have apropriate image for the device model.§
            + finished: `2009-11-10 24:00:00 +0000 UTC` (string, optional) - Deployment finish time
​
+ Response 404 (application/json)
    + Attributes (Error)
​
+ Response 500 (application/json)
    + Attributes (Error)
​

I'm interested in testing the 404 and 500 errors as well,  is the only way to split this in to separate request and responses?

**Comment by GregorioDiStefano** [comment]
Sill not getting anywhere, I've tried with the following:

```
### Deployment status [GET]
Check status for specified deployment

+ Parameters
    + id: `f81d4fae-7dec-11d0-a765-00a0c91e6bf6` (string,required) - Deployment identifier

+ Response 200 (application/json)
    + Attributes (DeploymentStatic)
        + id: `f81d4fae-7dec-11d0-a765-00a0c91e6bf6` (string)
        + finished: `2009-11-10 24:00:00 +0000 UTC` (string, optional) - Deployment finish time
        + status: `pending` (enum[string]) - Current deployment state
            + Members
                + `pending` - Pending to start
                + `inprogress` - In progress
                + `success` - Finished with success
                + `failure` - Failed

+ Request
    + id: `f81d4fae-7dec-11d0-a765-00000000000` (string,required) - Deployment identifier

+ Response 404 (application/json)
    + Attributes (Error)
```

but my original parameter f81d4fae-7dec-11d0-a765-00a0c91e6bf6 is being overwritten with the re-declared value f81d4fae-7dec-11d0-a765-00000000000.

**Comment by mhueter** [comment]
This feature would help clean up our docs.

We currently have to have empty `Request` headers to enable separate Dredd hooks for different responses like so:

+ Request Bad 
+ Response 400

+ Request Not Found
+ Response 404

This results in the warning:  `warn: Parser warning in file 'apiary.apib': (6) empty request message-body on line 689` 

This is only an issue for GET requests with different responses. Still, it would be less clumsy if Dredd could just check how many responses there are instead of insisting requests be paired.

tl;dr +1

**Comment by austinmcorso** [comment]
+1
